### PR TITLE
prod_name in get_par_by_name

### PR DESCRIPTION
**Pull request by dsavchenko**

### CODE REVIEW

File: cdci_data_analysis/analysis/instrument.py [hunk 1]
```diff
@@ -508,10 +508,17 @@ def get_html_draw(self, prod_name, image,image_header,catalog=None,**kwargs):
         return self.get_query_by_name(prod_name).get_html_draw( image,image_header,catalog=catalog,**kwargs)
 
     #def get_par_by_name(self,par_name, validate=False):
-    def get_par_by_name(self,par_name):
+    def get_par_by_name(self,par_name,add_src_query=True,add_instr_query=True,prod_name=None):
```
Comment (volodymyrss) [inline cdci_data_analysis/analysis/instrument.py:511]: space after coma?

**Comment by volodymyrss** [comment]
it seems to make sense, but could you please explain a little bit the motivation?

**Comment by dsavchenko** [comment]
> it seems to make sense, but could you please explain a little bit the motivation?

I came up with this in the context of https://github.com/oda-hub/dispatcher-plugin-nb2workflow/issues/47. When I build a request to the backend, I want to be sure that the parameter I'm working with is coming from the proper product. If parameters with the same name are annotated differently in the notebooks, they are not equivalent in the dispatcher. It addresses the TODO comment https://github.com/oda-hub/dispatcher-app/blob/439af5dc3a94c7d328f46445b0e464de6b1fdd60/cdci_data_analysis/analysis/instrument.py#L524 for this case.

It's not a strictly necessary change, I guess, as the problem in https://github.com/oda-hub/dispatcher-plugin-nb2workflow/issues/47 concerns SourceQuery parameters, but it's cleaner to use it this way

**Review by volodymyrss — APPROVED** [review]

**Comment by volodymyrss** [comment]
something breaks though

**Comment by volodymyrss** [comment]
oh, do we already have test for this? could you add one?

File: cdci_data_analysis/analysis/instrument.py [hunk 2]
```diff
@@ -504,24 +504,29 @@ def check_instrument_query_role(self, query_obj, product_type, roles, par_dic):
...
+        return self.get_query_by_name(prod_name).get_html_draw(image, image_header, catalog=catalog, **kwargs)
 
-    #def get_par_by_name(self,par_name, validate=False):
-    def get_par_by_name(self,par_name):
+    def get_par_by_name(self, par_name, add_src_query=True, add_instr_query=True, prod_name=None):
         p=None
-
+        
         for _query in self._queries_list:
+            if isinstance(_query, SourceQuery) and add_src_query==False:
```
Comment (burnout87) [inline cdci_data_analysis/analysis/instrument.py:514]:
```suggestion
            if isinstance(_query, SourceQuery) and not add_src_query:
```

File: cdci_data_analysis/analysis/instrument.py [hunk 3]
```diff
@@ -504,24 +504,29 @@ def check_instrument_query_role(self, query_obj, product_type, roles, par_dic):
...
+                continue
+            
+            if isinstance(_query, InstrumentQuery) and add_instr_query==False:
```
Comment (burnout87) [inline cdci_data_analysis/analysis/instrument.py:517]:
```suggestion
            if isinstance(_query, InstrumentQuery) and not add_instr_query:
```

File: cdci_data_analysis/analysis/instrument.py (hunk 2 above)
Comment (burnout87) [inline cdci_data_analysis/analysis/instrument.py:514]:
```suggestion
            if isinstance(_query, SourceQuery) and not add_src_query:
```

**Review by burnout87 — CHANGES_REQUESTED** [review]
Very useful thanks

**Review by burnout87 — APPROVED** [review]

**Review by volodymyrss — APPROVED** [review]

**Comment by dsavchenko** [comment]
could you merge please?

**Comment by volodymyrss** [comment]
I add you to maintainers, I think you add enough code. I merge though

**Pull request merged**